docs/src/scenarios/scenario2.md

```diff
@@ -3,31 +3,89 @@
+@variables T(t)=0.0 η(t)=0.0 cumulative_I(t)=0.0
+@parameters β₁=0.6 β₂=0.143 β₃=0.055 α=0.003 γ₁=0.007 γ₂=0.011 δ=0.1 μ=0.14
+eqs = [T ~ S + E + I + R + H + D
+       η ~ (β₁ * E + β₂ * I + β₃ * H) / T
+       Dₜ(S) ~ -η * S
+       Dₜ(E) ~ η * S - α * E
+       Dₜ(I) ~ α * E - (γ₁ + δ) * I
+       Dₜ(cumulative_I) ~ I
+       Dₜ(R) ~ γ₁ * I + γ₂ * H
+       Dₜ(H) ~ δ * I - (μ + γ₂) * H
+       Dₜ(D) ~ μ * H];
+@named seirhd = ODESystem(eqs)
+seirhd = structural_simplify(seirhd)
+prob = ODEProblem(seirhd, [], (0, 60.0), saveat = 1.0)
+sol = solve(prob)
+plot(sol)
+
+data = [I => sol[I], R => sol[R], H => sol[H], D => sol[D]]
 ```
 
 ## Model Analysis
 
 > Parameterize model either using data from the previous two months (October 28th – December 28th, 2021), or with relevant parameter values from the literature.
 
-```julia
-fit = datafit(prob, p_init, tsave, data)
+```@example scenario2
+p_priors = [β₁=>Normal(0.6, 0.1), β₂=>Normal(0.143, 0.01), β₃=>Normal(0.055, 0.01), α=>Normal(0.003, 0.001), γ₁=>Normal(0.007, 0.001), γ₂=>Normal(0.011, 0.01), δ=>Normal(0.1, 0.05), μ=>Normal(0.14, 0.05)]
+tsave = collect(0.0:1.0:2*30)
+fit = bayesian_datafit(prob, p_priors, tsave, data)
+prob = remake(prob; u0 = sol[end, :], p = Pair.(getfield.(fit, :first), mean.(getfield.(fit, :second))))
 ```
 
 ### Question 1
 
 > Forecast Covid cases and hospitalizations over the next 3 months under no interventions.
 
+```@example scenario2
+forecast_threemonths = solve(prob, tspan = (0.0, 90.0), saveat = 1.0)
+```
+
 ### Question 2
 
 > Based on the forecast, do we need interventions to keep total Covid hospitalizations under a threshold of 3000 on any given day? If there is uncertainty in the model parameters, express the answer probabilistically, i.e., what is the likelihood or probability that the number of Covid hospitalizations will stay under this threshold for the next 3 months without interventions?
 
+```@example scenario2
+need_intervention = any(x -> x > 3000, forecast_threemonths[H])
+
+posterior = Pair.(getfield.(fit,:first),Normal.(mean.(getfield.(fit,:second)),var.(getfield.(fit,:second))
+prob_violating_treshold(prob, posterior, [H>3000])
+```
+
 ### Question 3
 
 > Assume a consistent policy of social distancing/masking will be implemented, resulting in a 50% decrease from baseline transmission. Assume that we want to minimize the time that the policy is in place, and once it has been put in place and then ended, it can't be re-implemented. Looking forward from “today’s” date of Dec. 28, 2021, what are the optimal start and end dates for this policy, to keep projections below the hospitalization threshold over the entire 3-month period? How many fewer hospitalizations and cases does this policy result in?
 
+```@example scenario2
+#root_eqs = [H ~ 3000] # or 80% of 3000 
+#affect   = [η ~ η/2]
+#@named mask_system = ODESystem(eqs, t; continuous_events = root_eqs => affect)
+
+start_intervention = (t == tstart) => [η ~ η/2]
```

Comment:
these have to be defined inside the constraint, otherwise tstart and tstop don't exit.